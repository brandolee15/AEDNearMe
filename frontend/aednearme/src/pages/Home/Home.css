body {
    margin: 0 auto;
    background-color: #ede8f9 !important;
    padding: 20px, 20px, 20px, 20px !important;
}
