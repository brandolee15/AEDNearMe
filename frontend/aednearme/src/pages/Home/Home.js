import React from 'react'
import { useNavigate } from 'react-router';
import { Button } from '@mui/material';
import MainMap from '../../components/MainMap/MainMap';

const Home = () => {
  const navigate = useNavigate();

  const goToCPR = () => {
    navigate('/cpr')
  }

  const goToChoking = () => {
    navigate('/choking')
  }
  
  return (
    <>
    <div><MainMap /></div>
    <Button variant="contained" onClick={goToCPR}>CPR</Button>
    <Button variant="contained" onClick={goToChoking}>Choking</Button>
    <p>If in doubt, please call 999 or 112 for the emergency services</p>

    <p>If you suspect someone is having a cardiac arrest, call 999 or 112, and have someone track down the closest AED using the map. Follow the steps on the CPR page for instructions</p>
    </>
  )
}

export default Home