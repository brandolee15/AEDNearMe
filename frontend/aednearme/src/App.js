import React from "react";
import { Route, Routes, Outlet } from 'react-router-dom';
import Home from "./pages/Home/Home";
import { NavBar } from "./components/NavBar/NavBar";
import { Footer } from "./components/Footer/Footer.js";


function App() {
  return (
    <div className="App">
      <NavBar />
      <Routes>
          <Route path='/' element={<Home />} />
          <Route path='*' element={<p>nothing to see here, move on mate.</p>} />
      </Routes>

      <Footer/>

      <Outlet />
    </div>
  );
}

export default App;
