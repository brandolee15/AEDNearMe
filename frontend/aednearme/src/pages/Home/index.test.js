import { default as Home } from './Home';
import App from '../../App'
import { screen, render, cleanup } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import {MemoryRouter, Router, BrowserRouter} from 'react-router-dom'
import ReactDOM from 'react-dom';
import { createMemoryHistory } from 'history'






describe('Home Page', () => {
    let setPlayersMock;
    let handleChangeMock;

    // beforeEach(() => {
    //     // user = jest.fn();
    //     setPlayersMock = jest.fn();
    //     // render(<MemoryRouter><Home /></MemoryRouter>);
    // });

    afterEach(cleanup)

    test('it renders a heading', () => {
        render(<MemoryRouter><Home /></MemoryRouter>);
        let heading = screen.getByRole('heading');
        expect(heading).toBeInTheDocument();;
    });


      // Rendering

  it('renders welcome message', () => {
    render(<Home />, { wrapper: MemoryRouter })
    expect(screen.getByText('IN A LIFE THREATENING EMERGENCY, CALL 999')).toBeInTheDocument();
  });








  // Snapshot

  it('should take a snapshot', () => {
    // render(<Home />, { wrapper: MemoryRouter })
    const { asFragment } = render(<Home />, { wrapper: MemoryRouter })
    
    expect(asFragment(<Home />)).toMatchSnapshot()
  })




});