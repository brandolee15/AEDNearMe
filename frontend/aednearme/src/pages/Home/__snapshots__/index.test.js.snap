// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Page should take a snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiContainer-root pb-5 MuiContainer-maxWidthMd"
    style="padding-bottom: 100px;"
  >
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-4"
    >
      <div
        class="MuiGrid-root px-0 mx-auto d-flex align-items-center MuiGrid-item MuiGrid-grid-lg-12"
      >
        <div
          class="text-center"
        >
          <div>
            <h1>
              IN A LIFE THREATENING EMERGENCY, CALL 999
            </h1>
            <p>
              If you believe a person has suffered a sudden cardiac arrest (SCA) and is unresponsive and not breathing, immediately ring either 999 or 112 and ask for the ambulance service. They will advise you on what action to take and either inform you where your nearest defibrillator is located or ask if you have a defibrillator. The operator will remain on the line and ask you a series of questions which should only take you a few moments and advise you on what steps to take to assist in emergency.
            </p>
            <div />
            <div>
              Loading...
            </div>
            <div>
              <button
                class="MuiButtonBase-root MuiButton-root MuiButton-contained m-2 py-3 px-5 MuiButton-containedPrimary MuiButton-containedSizeLarge MuiButton-sizeLarge"
                tabindex="0"
                title="View Carolina React Admin Dashboard with Material-UI Free Live Preview"
                type="button"
              >
                <span
                  class="MuiButton-label"
                >
                  How to use AED
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </button>
              <button
                class="MuiButtonBase-root MuiButton-root MuiButton-contained m-2 py-3 px-5 MuiButton-containedPrimary MuiButton-containedSizeLarge MuiButton-sizeLarge"
                tabindex="0"
                title="View Carolina React Admin Dashboard with Material-UI Free Live Preview"
                type="button"
              >
                <span
                  class="MuiButton-label"
                >
                  Add a NEW location
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </button>
            </div>
            <small
              class="d-block pt-4"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
